src/DotLiquid/Tags/RegisterFilters.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using System;
+using DotLiquid.Exceptions;
+using DotLiquid.Util;
+
+namespace DotLiquid.Tags
+{
+    /// <summary>
+    /// Adds a Filter class containing one or more non-standard filters to the current context.
+    /// Duplicate/Additional calls to register the same filter class are benign and will be ignored. 
+    ///
+    /// {%- register_filters 'DotLiquid.ShopifyFilters' -%}
+    /// </summary>
+    public class RegisterFilters : Tag
+    {
+        private static readonly Regex Syntax = R.B(R.Q(@"\s*(.+)\s*"));
+
+        private string className;
+        private Type classType;
+
+        /// <inheritdoc/>
+        public override void Initialize(string tagName, string markup, List<string> tokens)
+        {
+            Match syntaxMatch = Syntax.Match(markup);
+            if (syntaxMatch.Success)
+                className = syntaxMatch.Groups[1].Value.Trim(new[] { '\'', '"' });
+
+            // The class name must be non-empty
+            if (string.IsNullOrWhiteSpace(className))
+                throw new SyntaxException(Liquid.ResourceManager.GetString("SimpleTagSyntaxException"), new[] { tagName });
+
+            // Attempt to find the class type
+            classType = Type.GetType(className);
```

Comment:
This is a security vulnerability. Potentially a template designer can access any (public?) class in .NET and call any method within that accepts one or more parameters. Without a dev specifying a whitelist of registerable filter classes (Shopify can be added by default to that whitelist) - and in that case we don't need reflection.

Reply:
Good point a white list makes sense, I've added one by default containing ShopifyFilters.

If this PR is approved I'll update the wiki with a section describing how to use `addfilters` including a developer section on how to whitelist new filters.